File: design-patterns/java/composite/filesystem/FileExplorerApp.java
public class FileExplorerApp {
    public static void main(String[] args) {
        FileSystemItem file1 = new File("readme.txt", 5);
        FileSystemItem file2 = new File("photo.jpg", 1500);
        FileSystemItem file3 = new File("data.csv", 300);

        Folder documents = new Folder("Documents");
        documents.addItem(file1);
        documents.addItem(file3);

        Folder pictures = new Folder("Pictures");
        pictures.addItem(file2);

        Folder home = new Folder("Home");
        home.addItem(documents);
        home.addItem(pictures);

        System.out.println("---- File Structure ----");
        home.printStructure("");

        System.out.println("\nTotal Size: " + home.getSize() + " KB");

        System.out.println("\n---- Deleting All ----");
        home.delete();
    }
}

File: design-patterns/python/composite/filesystem/file_explorer_app.py
from .folder import Folder
from .file import File

def main():
    # Create a file system structure
    root = Folder("Root")
    
    documents = Folder("Documents")
    pictures = Folder("Pictures")
    music = Folder("Music")
    
    # Add files to Documents
    doc1 = File("report.txt", 1024)
    doc2 = File("presentation.pptx", 2048)
    documents.add_item(doc1)
    documents.add_item(doc2)
    
    # Add files to Pictures
    pic1 = File("vacation.jpg", 3072)
    pic2 = File("family.png", 4096)
    pictures.add_item(pic1)
    pictures.add_item(pic2)
    
    # Add files to Music
    song1 = File("song1.mp3", 5120)
    song2 = File("song2.mp3", 6144)
    music.add_item(song1)
    music.add_item(song2)
    
    # Add folders to root
    root.add_item(documents)
    root.add_item(pictures)
    root.add_item(music)
    
    # Display file system structure
    print("File System Structure:")
    print(root)
    print(documents)
    print(pictures)
    print(music)
    
    # Display total size
    print(f"\nTotal Size: {root.get_size()} bytes")

if __name__ == "__main__":
    main() 